oio/admin/agent.py

```diff
@@ -0,0 +1,570 @@
+
+        self.log.info("Starting blob mover on %s with config %s",
+                      config['volume'],
+                      ",".join(["%s=%s" % (k, v) for k, v in config.items()]))
+        try:
+            logger = BlobStatsLogger(self.log,
+                                     stats.get("success"),
+                                     stats.get("fail"),
+                                     stats.get("bytes"))
+            worker = self.cm.blob_mover(config, logger, config['volume'])
+            worker.mover_pass()
+        except Exception as e:
+            self.log.error("Blob mover failed with error: %s" % format_exc(e))
+        self.client.unlock_score(dict(type="rawx", addr=src))
+        control.get('status').value = 2
+        control.get('end').value = int(time.time())
+
+    def check_running(self, vol):
+        """
+        Check if a mover job is already running on the specified volume
+        """
+        for _, job in self.jobs.items():
+            if job.get("host") == self.host and\
+                job['config'].get("volume") == vol\
+                    and job['control'].get("end").value == 0:
+                return True
+
+    def volume(self, type_, src):
+        """
+        Resolve the volume for the specified service
+        """
+        for svc in self.client.all_services(type_):
+            tags = svc.get("tags", {})
+            location = tags.get("tag.loc")
+            if svc.get("addr") == src and self.host in location.split("."):
+                return tags.get("tag.vol")
+        return None
+
+    def excluded(self, type_, exclude):
+        """
+        Resolve excluded rawx services from the exclude list
+        """
+        to_exclude = []
+        services = self.client.all_services(type_)
+
+        for excl in exclude:
+            incl_ = False
+            if excl.startswith("re:"):
+                excl = excl.split("re:", 1)[1]
+                if excl.startswith("!"):
+                    # Include instead of exclude
+                    incl_ = True
+                    excl = excl[1:]
+                loc = re.compile(excl)
+                for svc in services:
+                    tags_loc = svc.get("tags", {}).get("tag.loc", "")
+                    if (incl_ ^ bool(loc.match(tags_loc))):
+                        to_exclude.append(svc.get("addr"))
+            else:
+                for svc in services:
+                    for excl in exclude:
+                        if excl in svc.get("tags", {}).get("tag.loc", []):
+                            to_exclude.append(svc.get("addr"))
+            return to_exclude
+
+    def chunk_bases(self, bases, into=1):
+        """
+        Chunk meta2 bases into N chunks to allow
+        for mover parallelization
+        """
+        chunks = [bases[i::into] for i in range(into)]
+        for i, c in enumerate(chunks):
+            chunks[i] = [(k, v) for k, v in dict(c).iteritems()]
```

Comment:
I don't know which type is `bases`, but I suppose this can be simplified like
```suggestion
            chunks[i] = dict(c).items()
```